src/main/scala/com/microsoft/hyperspace/util/ResolverUtils.scala

```diff
@@ -71,4 +76,68 @@ object ResolverUtils {
       availableStrings: Seq[String]): Option[Seq[String]] = {
     Some(requiredStrings.map(resolve(spark, _, availableStrings).getOrElse { return None }))
   }
+
+  /**
+   * Finds all resolved strings for requiredStrings, from the given logical plan. Returns
+   * optional seq of resolved strings if all required strings are resolved, otherwise None.
+   *
+   * @param spark Spark session.
+   * @param requiredStrings List of strings to resolve.
+   * @param plan Logical plan to resolve against.
+   * @return Optional Seq of resolved strings if all required strings are resolved. Else, None.
+   */
+  def resolve(
+      spark: SparkSession,
+      requiredStrings: Seq[String],
+      plan: LogicalPlan): Option[Seq[String]] = {
+
+    def fixParts(parts: Seq[String], outputs: Seq[Attribute]): Seq[String] = {
+      var newParts = Seq.empty[String]
+      val h :: t = parts.toList
+      val topLevelField = outputs.find(_.name.compareToIgnoreCase(h) == 0)
+      topLevelField match {
+        case Some(o) =>
+          newParts = newParts :+ o.name
+          var children = o.dataType.asInstanceOf[StructType]
+          t.foreach { e =>
```

Comment:
how about `foldLeft(children) { (part, children)  => ... }`

Reply:
I don't know how that would go. We don't fold anything. For each part we need to find its nested counter part inside the struct nested layers.

Given `nesTED, LeaF, CNT` parts and the nested top level column with 
```
nested: struct (nullable = true)
 |-- leaf: struct (nullable = true)
 |    |-- v: integer (nullable = true)
 |    |-- cnt: integer (nullable = true)
 |    |-- other: string (nullable = true)
```
The `nesTED` will be resolved by finding the top level column name and using that. 

The `LeaF` will go one level into the struct search for it case insensitively and use the name of the struct.

For `CNT` it will go one more level into it the column search case insensitively for it out of the 3 leaves and use the name of the found struct.

Reply:
yep I meant for tail part, something like:

```scala
val (fixedParts, _) = t.foldLeft(Seq(o.name), children) => { (part, fixedParts, children) =>
      val elIdx: Int = children.fieldNames.indexWhere { f => f.compareToIgnoreCase(e) == 0 }
      if (elIdx >= 0) {
         // When found add it to the new set of fixed parts
          val fixedPartName = children.fieldNames(elIdx)
          chilchildren.fields.toSeq(elIdx).dataType match {
              case s: StructType => // et it up for the next part
                  (fixedParts :+ fixedPartName, s) 
                case _ => // Nothing to do, cannot go deeper, it must be a leaf
                  (fixedParts :+ fixedPartName, children)
              }
        } else { (fixedParts: + part, children) // or exception? }
}
```

Reply:
IMHO your suggestion is harder to read and understand than what I proposed. But I'm willing to change it. 

@imback82 WDYT? 

I'm asking you because I don't want to be in a position to change it back or to something else yet once more.